src/porepy/models/solution_strategy.py

```diff
@@ -440,9 +440,11 @@ def after_nonlinear_convergence(self) -> None:
 
         """
         solution = self.equation_system.get_variable_values(iterate_index=0)
-        self.equation_system.shift_time_step_values()
+        self.equation_system.shift_time_step_values(
+            max_index=len(self.time_step_indices)
```

Comment:
Added this functionality to avoid flooding the memory.
Same in line 430

Reply:
Seems like a very good idea!